### PR TITLE
Doc 11124: Release notes cbl 3.0.12

### PR DESCRIPTION
**Pull request by franrob-projects**
Release notes for 3.0.12 CBL:

- Android
- .Net
- Java
- Javascript
- Objective-C
- Swift

[DOC-11124](https://issues.couchbase.com/browse/DOC-11124) and [CBL 3.0.12](https://issues.couchbase.com/issues/?jql=project%20%3D%20CBL%20AND%20fixVersion%20%3D%203.0.12%20and%20level%20%3D%20%22Public%22)

### CODE REVIEW

File: modules/c/pages/releasenotes.adoc [hunk 1]
```diff
@@ -13,8 +13,5 @@ include::partial$_set_page_context_for_c.adoc[]
 include::{root-partials}_show_page_header_block.adoc[]
 
 [#maint-latest]
-include::partial$release-notes/couchbase-mobile-c-release-note.3.0.11.adoc[]
-include::partial$release-notes/couchbase-mobile-c-release-note.3.0.10.adoc[]
```
Comment (osfameron) [inline modules/c/pages/releasenotes.adoc:17]: why are you deleting these 2 release notes?

File: modules/c/partials/release-notes/couchbase-mobile-c-release-note.3.0.12.adoc [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4573[CBL-4573 - Generate API Doc]
+
+* https://issues.couchbase.com/browse/CBL-4561[CBL-4561 - Implement AcceptParentDomainCookies API]
+
+* https://issues.couchbase.com/browse/CBL-4489[CBL-4489 - Build XCFramework for iOS with Bitcode disabled]
+
+* https://issues.couchbase.com/browse/CBL-4531[CBL-4531 - Update PublicKey+Apple to use non deprecated keychain APIs]
+
+* https://issues.couchbase.com/browse/CBL-4167[CBL-4167 - SQL++ : COLLATE does not have a way to specify locale for UNICODE]
```
Comment (osfameron) [inline modules/c/partials/release-notes/couchbase-mobile-c-release-note.3.0.12.adoc:16]: does the `SQL++` render OK?
Notice in previous release notes how we tend to put the text bracketed inside `[++ ... ++]` which prevents most interpolation... but this is a literal ++ so even that wouldn't help.
Check the output!

File: modules/csharp/pages/releasenotes.adoc [hunk 3]
```diff
@@ -9,5 +9,5 @@ ifdef::prerelease[:page-status: {prerelease}]
 include::partial$_set_page_context_for_csharp.adoc[]
 
 [#maint-latest]
-include::partial$release-notes/couchbase-mobile-csharp-release-note.3.0.8.adoc[]
```
Comment (osfameron) [inline modules/csharp/pages/releasenotes.adoc:12]: Why are you deleting this release note?

File: modules/java/pages/releasenotes.adoc [hunk 4]
```diff
@@ -9,6 +9,8 @@ ifdef::prerelease[:page-status: {prerelease}]
 include::partial$_set_page_context_for_java.adoc[]
 
 [#maint-latest]
+
+include::partial$release-notes/couchbase-mobile-java-release-note3.0.12.adoc[]
```
Comment (osfameron) [inline modules/java/pages/releasenotes.adoc]: See! here you've done it right! Added the include above the existing ones 👍 

... although your naming convention is a bit wibbly, should be `couchbase-mobile-java-release-note.3.0.12.adoc`
e.g. with a dot before the 3.0.12
Ah I see, you're being consistent with the previous example which somebody (probably me!) got wrong.
OK, anyway, not super-important, but consistency is nice...

File: modules/java/partials:release-notes/couchbase-mobile-java-release-note3.0.12.adoc [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+Version 3.0.12 for {param-title} delivers the following features and enhancements:
+
+=== Enhancements
+
+* https://issues.couchbase.com/browse/CBL-4562[CBL-4562 - Implement AcceptParentDomainCookies API]
+
+* https://issues.couchbase.com/browse/CBL-4531[CBL-4531 - Update PublicKey+Apple to use non deprecated keychain APIs]
+
+* https://issues.couchbase.com/browse/CBL-4167[CBL-4167 - SQL++ : COLLATE does not have a way to specify locale for UNICODE]
```
Comment (osfameron) [inline modules/java/partials:release-notes/couchbase-mobile-java-release-note3.0.12.adoc]: Again here and elsewhere check the `++` renders and replace with `{sqlpp}` if not

File: modules/objc/partials/release-notes/couchbase-mobile-objc-release-note.3.0.12.adoc [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4570[CBL-4570 - URLEndpointListener.getURLs returns an empty list on Android v>=11]
+
+
+
+=== Known Issues
+
+
+
+* https://issues.couchbase.com/browse/CBL-637[++CBL-637 — Java Console app doesn't exit++]
```
Comment (osfameron) [inline modules/objc/partials/release-notes/couchbase-mobile-objc-release-note.3.0.12.adoc:60]: Ah, here you've got the `[++ ... ++]` pattern! This is an "inline pass-through macro" 
https://docs.asciidoctor.org/asciidoc/latest/pass/pass-macro/

**Review by osfameron — CHANGES_REQUESTED** [review]
Couple of things to check, but looks almost there

File: modules/c/pages/releasenotes.adoc (hunk 1 above)
Comment (franrob-projects) [reply modules/c/pages/releasenotes.adoc:17]: added back on a new commit

File: modules/c/partials/release-notes/couchbase-mobile-c-release-note.3.0.12.adoc (hunk 2 above)
Comment (franrob-projects) [reply modules/c/partials/release-notes/couchbase-mobile-c-release-note.3.0.12.adoc:16]: This seems to be rendering ok on antora build

File: modules/csharp/pages/releasenotes.adoc (hunk 3 above)
Comment (franrob-projects) [reply modules/csharp/pages/releasenotes.adoc:12]: re-added in next commit

File: modules/java/pages/releasenotes.adoc (hunk 4 above)
Comment (franrob-projects) [reply modules/java/pages/releasenotes.adoc]: added '.' in nect commit

**Review by osfameron — APPROVED** [review]
👍

File: modules/android/partials/release-notes/couchbase-mobile-android-release-note.3.0.12.adoc [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+
+[#maint-3-0-12]
+== 3.0.12 -- June 2023
+
+Version 3.0.12 for {param-title} delivers the following features and enhancements:
+
+=== Enhancements
+
+* https://issues.couchbase.com/browse/CBL-4531[CBL-4531 - Update PublicKey+Apple to use non deprecated keychain APIs]
```
Comment (bfavini) [inline modules/android/partials/release-notes/couchbase-mobile-android-release-note.3.0.12.adoc:12]: PublicKey+Apple now uses current keychain APIs.

**Review by bfavini — APPROVED** [review]

File: modules/android/partials/release-notes/couchbase-mobile-android-release-note.3.0.12.adoc [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4167[CBL-4167 - SQL++ : COLLATE does not have a way to specify locale for UNICODE]
+
+* https://issues.couchbase.com/browse/CBL-4024[CBL-4024 - Change BuiltInWebSocket to do preemptive auth instead of challenge auth by default (Port)]
+
+* https://issues.couchbase.com/browse/CBL-3903[CBL-3903 - PORT - Assertion failure when stopping replicator while replicator is connecting]
```
Comment (djpongh) [inline modules/android/partials/release-notes/couchbase-mobile-android-release-note.3.0.12.adoc]: Can you remove the "PORT -"

File: modules/c/partials/release-notes/couchbase-mobile-c-release-note.3.0.12.adoc [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4024[CBL-4024 - Change BuiltInWebSocket to do preemptive auth instead of challenge auth by default (Port)]
+
+* https://issues.couchbase.com/browse/CBL-3903[CBL-3903 - PORT - Assertion failure when stopping replicator while replicator is connecting]
```
Comment (djpongh) [inline modules/c/partials/release-notes/couchbase-mobile-c-release-note.3.0.12.adoc]: Remove "PORT"

File: modules/csharp/partials/release-notes/couchbase-mobile-csharp-release-note.3.0.12.adoc [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4550[CBL-4550 - Make IReachability interface public]
+
+* https://issues.couchbase.com/browse/CBL-4531[CBL-4531 - Update PublicKey+Apple to use non deprecated keychain APIs]
+
+* https://issues.couchbase.com/browse/CBL-4167[CBL-4167 - SQL++ : COLLATE does not have a way to specify locale for UNICODE]
+
+* https://issues.couchbase.com/browse/CBL-4024[CBL-4024 - Change BuiltInWebSocket to do preemptive auth instead of challenge auth by default (Port)]
+
+* https://issues.couchbase.com/browse/CBL-3903[CBL-3903 - PORT - Assertion failure when stopping replicator while replicator is connecting]
```
Comment (djpongh) [inline modules/csharp/partials/release-notes/couchbase-mobile-csharp-release-note.3.0.12.adoc]: Remove "PORT"

File: modules/java/partials/release-notes/couchbase-mobile-java-release-note3.0.12.adoc [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4562[CBL-4562 - Implement AcceptParentDomainCookies API]
+
+* https://issues.couchbase.com/browse/CBL-4531[CBL-4531 - Update PublicKey+Apple to use non deprecated keychain APIs]
+
+* https://issues.couchbase.com/browse/CBL-4167[CBL-4167 - SQL++ : COLLATE does not have a way to specify locale for UNICODE]
+
+* https://issues.couchbase.com/browse/CBL-4024[CBL-4024 - Change BuiltInWebSocket to do preemptive auth instead of challenge auth by default (Port)]
+
+* https://issues.couchbase.com/browse/CBL-3903[CBL-3903 - PORT - Assertion failure when stopping replicator while replicator is connecting]
```
Comment (djpongh) [inline modules/java/partials/release-notes/couchbase-mobile-java-release-note3.0.12.adoc]: Remove "PORT"

File: modules/objc/partials/release-notes/couchbase-mobile-objc-release-note.3.0.12.adoc [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4487[CBL-4487 - Build release framework without bitcode enabled]
+
+* https://issues.couchbase.com/browse/CBL-4531[CBL-4531 - Update PublicKey+Apple to use non deprecated keychain APIs]
+
+* https://issues.couchbase.com/browse/CBL-4167[CBL-4167 - SQL++ : COLLATE does not have a way to specify locale for UNICODE]
+
+* https://issues.couchbase.com/browse/CBL-4024[CBL-4024 - Change BuiltInWebSocket to do preemptive auth instead of challenge auth by default (Port)]
+
+* https://issues.couchbase.com/browse/CBL-3903[CBL-3903 - PORT - Assertion failure when stopping replicator while replicator is connecting]
```
Comment (djpongh) [inline modules/objc/partials/release-notes/couchbase-mobile-objc-release-note.3.0.12.adoc]: Remove "PORT"

File: modules/swift/partials/release-notes/couchbase-mobile-swift-release-note.3.0.12.adoc [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+
+* https://issues.couchbase.com/browse/CBL-4487[CBL-4487 - Build release framework without bitcode enabled]
+
+* https://issues.couchbase.com/browse/CBL-4531[CBL-4531 - Update PublicKey+Apple to use non deprecated keychain APIs]
+
+* https://issues.couchbase.com/browse/CBL-4167[CBL-4167 - SQL++ : COLLATE does not have a way to specify locale for UNICODE]
+
+* https://issues.couchbase.com/browse/CBL-4024[CBL-4024 - Change BuiltInWebSocket to do preemptive auth instead of challenge auth by default (Port)]
+
+* https://issues.couchbase.com/browse/CBL-3903[CBL-3903 - PORT - Assertion failure when stopping replicator while replicator is connecting]
```
Comment (djpongh) [inline modules/swift/partials/release-notes/couchbase-mobile-swift-release-note.3.0.12.adoc]: Remove "PORT"

**Pull request merged**